lib/data_processor.rb

```diff
@@ -0,0 +1,66 @@
+require 'open-uri'
+require 'csv'
+
+class DataProcessor
+
+  def initialize(user_x, user_y, url)
```

Comment:
What would happen if we provided a file path instead of an URL?

Reply:
It was specified that the argument is an url. I can add support for local paths.